src/icrar/leap-accelerate/tests/math/MatrixTests.cc

```diff
@@ -20,40 +20,100 @@
  * MA 02111 - 1307  USA
  */
 
-#include <icrar/leap-accelerate/math/cuda/matrix.h>
-#include <icrar/leap-accelerate/math/casa/matrix_invert.h>
+#include <gtest/gtest.h>
+
+#include <icrar/leap-accelerate/tests/math/eigen_helper.h>
+
+#include <icrar/leap-accelerate/common/eigen_extensions.h>
 #include <icrar/leap-accelerate/math/cpu/matrix_invert.h>
 
 #include <Eigen/Core>
-#include <Eigen/LU>
-
-#include <icrar/leap-accelerate/tests/test_helper.h>
-#include <gtest/gtest.h>
+#include <Eigen/Sparse>
 
+#include <iostream>
 #include <array>
-#include <vector>
-#include <sstream>
 
-class matrix_tests : public testing::Test
+class MatrixTests : public testing::Test
 {
-    const double TOLERANCE = 0.0001; 
+    const double TOLERANCE = 0.0001;
+
 public:
-    matrix_tests()
+    MatrixTests() = default;
+
+    void SetUp() override
     {
+        //int deviceCount = 0;
+        //cudaGetDeviceCount(&deviceCount);
+        //ASSERT_NE(deviceCount, 0);
 
+        // See this page: https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__DEVICE.html
     }
 
-    void SetUp() override
+    void TearDown() override
     {
 
     }
```

Comment:
Aren't those not needed if they are not implemented?

Reply:
I think I'm meant to call cudaDeviceReset() here